packages/lib/src/vis/shared/ZoomSelectionMesh.tsx

```diff
@@ -0,0 +1,56 @@
+import { useThree } from '@react-three/fiber';
+
+import type { ModifierKey, SelectionPoints } from '../models';
+import SelectionMesh from './SelectionMesh';
+import SelectionRect from './SelectionRect';
+import { useMoveCameraTo } from './hooks';
+
+interface Props {
+  modifierKey?: ModifierKey;
+}
+
+function ZoomSelectionMesh(props: Props) {
+  const { modifierKey } = props;
+
+  const moveCameraTo = useMoveCameraTo();
+
+  const { width, height } = useThree((state) => state.size);
+  const camera = useThree((state) => state.camera);
+
+  const onSelectionEnd = (selection: SelectionPoints) => {
+    const { startPoint, endPoint } = selection;
+    if (startPoint.x === endPoint.x && startPoint.y === endPoint.y) {
```

Comment:
Not for this PR, but we could imagine calculating the width/height of the selection and returning when it's below a threshold -- to avoid zooming in on a tiny area by mistake.